src/cargo/core/compiler/standard_lib.rs

```diff
@@ -125,9 +125,12 @@ pub fn resolve_std<'gctx>(
     // now. Perhaps in the future features will be decoupled from the resolver
     // and it will be easier to control feature selection.
     let current_manifest = src_path.join("library/sysroot/Cargo.toml");
-    // TODO: Consider doing something to enforce --locked? Or to prevent the
-    // lock file from being written, such as setting ephemeral.
-    let mut std_ws = Workspace::new_virtual(src_path, current_manifest, virtual_manifest, gctx)?;
+    let mut std_ws =
+        Workspace::new_virtual(src_path.clone(), current_manifest, virtual_manifest, gctx)?;
+
+    // The source checkout isn't managed by us, so setting ephemeral here
+    // ensures the lockfile isn't updated.
+    std_ws.set_ephemeral(true);
```

Comment:
Could this be split out into a separate commit or even separate PR?  Seems like this is fairly important we shouldn't hide it away in these other changes